src/mapReducers.js

```diff
@@ -1,4 +1,5 @@
 import _ from 'lodash';
+import { isTargetAllAction } from './context';
```

Comment:
This discussion takes too long, we can comment live. I will just answer your last comment. 
Your reducer doesn't know about ALL, only mapReducer should know about it. Idea of mapReducer is to apply action with reducer only to substate that is under key returned by keySelector. That's it.